### DISCUSSION TITLE
Problems with DAC PCM5102 I2S

### DISCUSSION THREAD

**Discussion by eziosoma** [probonopd/MiniDexed#426] [Q&A] [answered]
![20230124_184709](https://user-images.githubusercontent.com/4121029/214369613-445cb489-9681-4438-a348-1df833cbff6d.jpg)
Hello,
I have a problem with the DAC PCM5102 I2S.
After having done a provisional wiring of the Raspy Pi, LCD, Encoder, midi module and DAC I was able to get everything running ok with a great sound.

I decided then to put everything in a box, so I disconnected all and put various component in a Box. After having re-wired all I was not able to get the DAC running again. I'm able to use PWM, but DAC does not want to work.
I have disconnected all, reconnected all at least four times. I changed the DAC, I changed the Raspi Pi, I changed the midi module. I suspected that the problem was related to MIDI so I put the Minidexed in midi debug mode and I was able to see on a screen the MIDI message correctly generated and received.

So my question is now: how I can debug this? I need an oscilloscope?

The only think that I have not changed is the Software part (SD card)

What do you suggest to do?

Many thanks for your attention. Any suggestion will be appreciated.

**Comment by probonopd** [comment, marked as answer]
Has your PCM5102 module ever produced any sound (e.g., not using MiniDexed)?
How are the solder bridges on the back side of the PCM5102 module configured, if there are any?

**Reply by eziosoma** [reply]
![20230124_203618](https://user-images.githubusercontent.com/4121029/214391373-02ada9fb-2e65-432f-9644-395827754037.jpg)
The module I am using now is brand new, just bougth from amazon. The previous one that is the one that worked with provisional wiring and stopped to work when I reassembly all in the box,  was destryed by my attempt of removing it from the box. 
The soldering on the back is the following: 
1 low 2 low 3 high 4 low

**Reply by probonopd** [reply]
Please measure with a multimeter that the setting is like it should be, and that there are no extra solder bridges. Sometimes these modules come wrong from the factory.

And please test it without MiniDexed, e.g., using Raspbian - to know for sure that the module itself is working.

**Reply by eziosoma** [reply]
I have tested with a multimeter that the settings are correct and there are no extra bridges.
This afternoon I will configure a Raspy Pi with Raspbian and the DAC to see if it works under other conditions. In the meanwhile I have ordered another DAC from another supplier, just in case.

**Reply by eziosoma** [reply]
I have tested the DAC on Raspbian following the instructions reported in this blog:  https://blog.himbeer.me/2018/12/27/how-to-connect-a-pcm5102-i2s-dac-to-your-raspberry-pi/
The device is correctly seen as per the command

aplay -l

That reports:

pi@raspberrypi:~ $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_dac], device 0: HifiBerry DAC HiFi pcm5102a-hifi-0 [HifiBerry DAC HiFi pcm5102a-hifi-0]
Subdevices: 1/1
Subdevice #0: subdevice #0
card 1: vc4hdmi [vc4-hdmi], device 0: MAI PCM i2s-hifi-0 [MAI PCM i2s-hifi-0]
Subdevices: 1/1
Subdevice #0: subdevice #0

But no sound is produced in the earphone connected to the DAC.

At this point I wait for the new one just ordered that should arrive tomorrow.

**Reply by diyelectromusic** [reply]
> I have tested with a multimeter that the settings are correct and there are no extra bridges.

I know its hard to tell in that photo, but are those all really bridged in the same configuration as described here? https://github.com/gmcn42/clumsyMIDI

H3L doesn't look connected across H at all in that photo...?

(sorry if I'm just restated the obvious)

Kevin

**Reply by eziosoma** [reply]
To exclude the obvious is always a good principle in debugging...

I know that the photo is not so good, but actually there are little smd resistors soldered as bridges (I assume of 1 ohm or less), I have also checked, triple chcked, with the multimeter and confiuration correspond.

**Reply by eziosoma** [reply]
New module arrived without bridges. I soldered them and connected to the Raspi + Raspian. It works. Now I have to reassemble the minidexed and seeing if I'm a lucky man...

**Reply by eziosoma** [reply]
Yes now it works! so the problem was a broken DAC module.
